src/seqn/trans.lisp

```diff
@@ -0,0 +1,399 @@
+                                            (= cadrobjnum 1))
+                                       car)
+                                      ((and (not (= carobjnum 1))
+                                            (= cadrobjnum 0))
+                                       (infix-creation :*
+                                                       ifzero
+                                                       carobj))
+                                      ((and (= carobjnum 1)
+                                            (= cadrobjnum 0))
+                                       ifzero)
+                                      ((and (= carobjnum cadrobjnum 1))
+                                       (infix-creation :+
+                                                       ifzero
+                                                       car))
+                                      (t output))))
+                             ((typep carobj 'vamp:constant)
+                              (let ((carobjnum (vamp:const carobj)))
+                                (cond ((= carobjnum 0) (infix-creation :*
+                                                                       car
+                                                                       cadrobj))
+                                      ((= carobjnum 1) (infix-creation :+
+                                                                       ifzero
+                                                                       ifone))
+                                      (t output))))
+                             ((typep cadrobj 'vamp:constant)
+                              (let ((cadrobjnum (vamp:const cadrobj)))
+                                (cond ((= cadrobjnum 0) (infix-creation :*
+                                                                        ifzero
+                                                                        carobj))
+                                      ((= cadrobjnum 1) (infix-creation :+
+                                                                        (infix-creation :*
+                                                                                        ifzero
+                                                                                        carobj)
+                                                                        car))
+                                      (t  output))))
+                             (t output))))
+         (prod-list left
+                    right)))))
+
+(defmethod to-vampir ((obj shift-front) inputs constraints)
+  "Takes the MCADR entry and moves it upward leaving everything
+else fixed. Note that we have to be careful as inputs will have 0es
+removed already and hence we cannot count as usual"
+  (declare (ignore constraints))
+  (append (cons (nth (1- (mcadr obj))
+                     inputs)
+                (subseq inputs 0 (1- (mcadr obj))))
+          (subseq inputs (mcadr obj))))
```

Comment:
let the mcadr as you use it 3 times.

also for uniformity reasons it might be good to do 1 append, with the following
```lisp
(append (list moved-element) initial rest)
```

would make the code obvious. Here I had to read it through a bit to catch it.